### PR TITLE
Switch fully to conversations api, use new fine-grained scopes and make logging consistent

### PR DESCRIPTION
**Pull request by ryan-lane**
This change switches fully to the conversations API, switches all slack api calls to bot token by default (using new fine-grained scopes for bots), and makes logging consistent by adding context to the bot and team.

### CODE REVIEW

File: omnibot/routes/api.py [hunk 1]
```diff
@@ -107,48 +107,48 @@ def slack_event():
         msg = 'No team_id in event.'
         logger.error(
             msg,
-            extra={'app': api_app_id},
+            extra={'bot_id': api_app_id},
```
Comment (johnisutanto) [inline omnibot/routes/api.py]: do we need to add this to keyword like you did last time? https://github.com/lyft/omnibot-private/blob/master/manifest.yaml#L100

File: omnibot/routes/api.py [hunk 2]
```diff
@@ -200,7 +200,10 @@ def slack_slash_command():
         )
         logger.error(
             msg,
-            extra={'team': team.team_id},
+            extra=merge_logging_context(
+                {'expected_team_id': team.team_id},
```
Comment (johnisutanto) [inline omnibot/routes/api.py]: same here for this [keyword](https://github.com/lyft/omnibot-private/blob/master/manifest.yaml#L100).

File: omnibot/routes/api.py [hunk 3]
```diff
@@ -200,7 +200,10 @@ def slack_slash_command():
         )
         logger.error(
             msg,
-            extra={'team': team.team_id},
+            extra=merge_logging_context(
```
Comment (johnisutanto) [inline omnibot/routes/api.py]: can we use unpacking here instead? 
```
{ **{'expected_team_id': team.team_id }, **bot.logging_context }
```

Comment (ryan-lane) [reply omnibot/routes/api.py]: I'd really prefer not to unpack in every spot. it's easier to read with the function, and it's a lot easier to have things be written correctly if we just need to call a function with dicts that returns a dict.

**Review by johnisutanto — DISMISSED** [review]

**Review by johnisutanto — APPROVED** [review]

**Pull request merged**